images/chaos-daemon/Dockerfile

```diff
@@ -6,23 +6,28 @@ ARG HTTP_PROXY
 ENV http_proxy $HTTP_PROXY
 ENV https_proxy $HTTPS_PROXY
 
-RUN apt-get update && apt-get install -y tzdata iptables ipset stress-ng iproute2 fuse util-linux procps curl && rm -rf /var/lib/apt/lists/*
+RUN apt-get update && \
+    apt-get install -y tzdata iptables ipset stress-ng iproute2 fuse util-linux procps curl && \
+    rm -rf /var/lib/apt/lists/*
 
 RUN update-alternatives --set iptables /usr/sbin/iptables-legacy
 
 ENV RUST_BACKTRACE 1
-
-RUN curl -L https://github.com/chaos-mesh/toda/releases/download/v0.2.0/toda-linux-amd64.tar.gz -o /usr/local/bin/toda.tar.gz
-RUN curl -L https://github.com/chaos-mesh/nsexec/releases/download/v0.1.5/nsexec-linux-amd64.tar.gz -o /usr/local/bin/nsexec.tar.gz
-RUN curl -L https://github.com/chaos-mesh/rs-tproxy/releases/download/v0.2.3/tproxy-linux-amd64.tar.gz -o /usr/local/bin/tproxy.tar.gz
-
-RUN tar xvf /usr/local/bin/toda.tar.gz -C /usr/local/bin
-RUN rm /usr/local/bin/toda.tar.gz
-RUN tar xvf /usr/local/bin/tproxy.tar.gz -C /usr/local/bin
-RUN rm /usr/local/bin/tproxy.tar.gz
-RUN tar xvf /usr/local/bin/nsexec.tar.gz -C /usr/local/bin
-RUN rm /usr/local/bin/nsexec.tar.gz
-RUN cp /usr/local/bin/libnsenter.so /usr/local/lib/libnsenter.so
+ARG TARGET_PLATFORM=amd64
+
+RUN curl -L https://github.com/chaos-mesh/toda/releases/download/v0.2.0/toda-linux-amd64.tar.gz | tar xz -C /usr/local/bin
+RUN case "$TARGET_PLATFORM" in \
+    'amd64') \
+    export NSEXEC_ARCH='x86_64'; \
+    ;; \
+    'arm64') \
+    export NSEXEC_ARCH='aarch64'; \
+    ;; \
+    *) echo >&2 "error: unsupported architecture '$TARGET_PLATFORM'"; exit 1 ;; \
+    esac; \
+    curl -L https://github.com/chaos-mesh/nsexec/releases/download/v0.1.6/nsexec-$NSEXEC_ARCH-unknown-linux-gnu.tar.gz | tar xz -C /usr/local/bin; \
+    cp /usr/local/bin/libnsenter.so /usr/local/lib/libnsenter.so
+RUN curl -L https://github.com/chaos-mesh/rs-tproxy/releases/download/v0.2.3/tproxy-linux-amd64.tar.gz | tar xz -C /usr/local/bin
```

Comment:
The newer version of `tproxy` also supports `aarch64`.

Reply:
updated!